gbpservice/nfp/configurator/api/v1/controllers/controller.py

```diff
@@ -215,3 +265,68 @@ def to_dict(self):
 
         """
         return {}
+
+
+""" CloudService keeps all information of uservice along with initialized
+    RPCClient object using which rpc is routed to over the cloud service.
+"""
+
+
+class CloudService():
+
+    def __init__(self, **kwargs):
+        self.service_name = kwargs.get('service_name')
+        self.topic = kwargs.get('topic')
+        self.host = kwargs.get('host')
+        self.reporting_interval = kwargs.get('reporting_interval')
+        self.rpcclient = RPCClient(topic=self.topic, host=self.host)
+
+
+"""RMQConsumer for over the cloud services.
+
+This class access rabbitmq's 'configurator-notifications' queue
+to pull all the notifications came from over the cloud services.
+"""
+
+
+class RMQConsumer(object):
+
+    def __init__(self, rabbitmq_host, queue):
+        self.rabbitmq_host = rabbitmq_host
+        self.queue = queue
+        self.create_connection()
+
+    def create_connection(self):
+        self.connection = pika.BlockingConnection(pika.ConnectionParameters(
+            host=self.rabbitmq_host))
+
+    def pull_notifications(self):
+        notifications = []
+        msgs_acknowledged = False
+        try:
+            self.channel = self.connection.channel()
+            self.queue_declared = self.channel.queue_declare(queue=self.queue,
+                                                             durable=True)
+            pending_msg_count = self.queue_declared.method.message_count
+            log = ('[notifications queue:%s, pending notifications:%s]'
+                   % (self.queue, pending_msg_count))
+            for i in range(pending_msg_count):
+                method, properties, body = self.channel.basic_get(self.queue)
+                notifications.append(ast.literal_eval(body))
+
+            # Acknowledge all messages delivery
+            if pending_msg_count > 0:
+                self.channel.basic_ack(delivery_tag=method.delivery_tag,
+                                       multiple=True)
+                msgs_acknowledged = True
+
+            self.channel.close()
+            return notifications
+        except pika.exceptions.ConnectionClosed:
+            self.create_connection()
```

Comment:
There is a possibility of indefinite recursion if there is an exception in create_connection. Why is this retried in exception?